pivotx/modules/module_tags.php

```diff
@@ -94,9 +93,25 @@ function getTagCosmos($max=0, $weblogname='', $match='', $exclude='') {
 function getTagCosmosFlat($max=0,$weblogname,$match,$exclude=array()) {
     global $PIVOTX;
 
+    
+    // Note that on Windows systems, filectime will show the file creation time
+    // So instead we encode the modification date in a separate file...
+    //$file_time = filectime( $PIVOTX['paths']['db_path']."ser_tags.php" );
+    $file_time = 0;
+    $time_stamp_filename = $PIVOTX['paths']['db_path']."ser_tags.php.log";
+    if( file_exists($time_stamp_filename) ){                
+        $file_time = (int)readAFile($time_stamp_filename);
+    }
+    
+    $current_time = time();
+    $time_out = 60 * $PIVOTX['config']->get('tag_cache_timeout');
+    $file_exists = file_exists($PIVOTX['paths']['db_path']."ser_tags.php");
+    $time_delta = $current_time - $file_time; 
+    
+
     // If the cached version is fresh enough, we restore that
-    if ( (file_exists($PIVOTX['paths']['db_path']."ser_tags.php"))  &&
-        (filectime($PIVOTX['paths']['db_path']."ser_tags.php") > (time() - (60 * $PIVOTX['config']->get('tag_cache_timeout') ))) ) {
```

Comment:
Not db related but triggered unnecessary file updates.

here we detect if ser_tags.php was recently updated using "filectime()", unfortunately it only works on unix systems since filectime() will not return the last updated time but the time of creation on windows platform. This is why we track the last updated time of that file in a separate file ser_tags.php**.log**